be/src/env/env.h

```diff
@@ -217,6 +217,9 @@ class RandomAccessFile {
     RandomAccessFile() = default;
     virtual ~RandomAccessFile() = default;
 
+    virtual Status open() { return Status::OK(); }
```

Comment:
not a fan of explicit initialize and finalize, we can initialize it in constructor, or if you want lazy initialization, maybe we can init it when first `read` happends

Reply:
If you just look at this piece of code, we can remove `open`. But from scanner like `hdfs_scanner.cpp`, there is a `open` interface, which is exactly for file open and initialization. 

```
Status HdfsScanner::open(RuntimeState* runtime_state) {
    if (_is_open) {
        return Status::OK();
    }
#ifndef BE_TEST
    RETURN_IF_ERROR(_scanner_params.fs->open());
#endif
    _build_file_read_param();
    auto status = do_open(runtime_state);
    if (status.ok()) {
        _is_open = true;
#ifndef BE_TEST
        (*_scanner_params.open_limit)++;
#endif
        LOG(INFO) << "open file success: " << _scanner_params.fs->file_name();
    }
    return status;
}
```

Reply:
`RandomAccessFile` is an abstract class, and it's using in many place, it's not just use by `HdfsScanner`. Because `HdfsScanner` has an `open` interface, we need to add an interface for `RandomAccessFile` too is not a strong enough reson.

Reply:
For example:
```
Status read_at(uint64_t offset, const Slice& result) const {
  if (!_init) {
    _open();
    _init = true;
  }
  ...
}
```

Reply:
OK. Will fix that.

Reply:
So we should remove `open` but keep `close`? Or remove both?